MainMovement.py

```diff
@@ -0,0 +1,163 @@
+            if port.vid == 1155 and port.pid == 22336:
+                self.ser = serial.Serial(port.device)  # open serial port
+                print("leidis seriali")
+                break
+        if self.ser == None:
+            raise Exception("Could not establish serial connection")
+
+    def close(self):
+        print("sulgesin seriali")
+        self.ser.close()
+
+    def sendCommand(self, thrower_speed, speed1, speed2, speed3):
+        self.ser.write(struct.pack('<hhhHBH', speed1, speed2, speed3, thrower_speed, self.disable_failsafe, 0xAAAA))
+        # reads as many bytes as there are to read
+        self.ser.read(self.ser.in_waiting)
+
+    def move(self, x_speed, y_speed, robotAngularVelocity, thrower_speed):
+        robotSpeed = math.sqrt(x_speed * x_speed + y_speed * y_speed)
+        robotDirectionAngle = math.atan2(y_speed, x_speed)
+        wheelSpeeds = [robotSpeed * math.cos(robotDirectionAngle - angle) +
+                       self.wheelDistanceFromCenter * robotAngularVelocity for angle in self.wheelAngles]
+        wheelSpeedsInMainboardUnits = [round(speed*self.wheelSpeedToMainboardUnits) for speed in wheelSpeeds]
+        # thrower_speed = 0 #temporary
+        # print(wheelSpeedsInMainboardUnits)
+        self.sendCommand(thrower_speed, wheelSpeedsInMainboardUnits[0], wheelSpeedsInMainboardUnits[1], wheelSpeedsInMainboardUnits[2])
+
+
+class States(Enum):
+    spin = 0
+    drive = 1
+    orbit = 2
+    throw = 3
+    read_ref = 4
+
+
+class StateMachine():
+    def __init__(self, motion):
+        self.motion = motion
+
+
+    def spin(self, processedData):
+        # if it finds more than 1 ball to choose from, it goes to the drive function
+        if len(processedData.balls) > 0:
+            #print(processedData.balls)
+            return States.drive
+        else:
+            # otherwise it spins until it finds some
+            self.motion.move(0, 0, 1, 0)
+        return States.spin
+
+    def drive(self, processedData):
+        y_keskmine = list()
+        if len(processedData.balls) > 0:
+            ball = processedData.balls[-1]
+            #print("distance: {}".format(ball))
+            xmiddif = (424 - (ball.x)) * 0.002              #approcing ball relative to the ball's x and y coordinate
+            yfor = (360 - (ball.distance)) * 0.005
+            #print("x: ", xmiddif, ", y: ", yfor)
+            self.motion.move(0, yfor, xmiddif, 0)
+            if ball.distance >= 350:
+                # if it is close enough to a ball, it should start orbiting around it to find the basket:
+                return States.orbit
+        # if it loses the ball for some reason (opponent takes it), it starts spinning to look for a new one:
+        else:
+            return States.spin
+        return States.drive
+
+    def orbit(self, processedData, basketColor):
+        
+        basket = processedData.basket_b if basketColor == 'b' else processedData.basket_m
```

Comment:
Random string defines basket color. Maybe use the Color enum?

Reply:
used color enum